src/main/org/codehaus/groovy/transform/stc/StaticTypeCheckingSupport.java

```diff
@@ -1116,39 +1117,23 @@ private static ClassNode makeRawType(final ClassNode receiver) {
             for (int j=i+1;j<list.size();j++) {
                 MethodNode two = list.get(j);
                 if (toBeRemoved.contains(two)) continue;
-                if (one.getName().equals(two.getName()) && one.getDeclaringClass()==two.getDeclaringClass()) {
-                    Parameter[] onePars = one.getParameters();
-                    Parameter[] twoPars = two.getParameters();
-                    if (onePars.length == twoPars.length) {
-                        boolean sameTypes = true;
-                        for (int k = 0; k < onePars.length; k++) {
-                            Parameter onePar = onePars[k];
-                            Parameter twoPar = twoPars[k];
-                            if (!onePar.getType().equals(twoPar.getType())) {
-                                sameTypes = false;
-                                break;
-                            }
-                        }
-                        if (sameTypes) {
-                            ClassNode oneRT = one.getReturnType();
-                            ClassNode twoRT = two.getReturnType();
-                            if (oneRT.isDerivedFrom(twoRT) || oneRT.implementsInterface(twoRT)) {
-                                toBeRemoved.add(two);
-                            } else if (twoRT.isDerivedFrom(oneRT) || twoRT.implementsInterface(oneRT)) {
-                                toBeRemoved.add(one);
-                            }
+                Parameter[] onePars = one.getParameters();
+                Parameter[] twoPars = two.getParameters();
+                if (onePars.length == twoPars.length) {
+                    if (areOverloadMethodsInSameClass(one,two)) {
+                        if (ParameterUtils.parametersEqual(onePars, twoPars)){
```

Comment:
If the `Parameter[]` parameters are removed from `areEquivalentInterfaceMethods` method I think these arguments can be replaced with `one.getParameters(), two.getParameters()` and then the local variables wouldn't be needed.  Just a suggestion.